enterprise/api/v1/views.py

```diff
@@ -289,14 +392,9 @@ def retrieve(self, request, pk=None):  # pylint: disable=invalid-name
         Returns:
             (Response): DRF response object containing course catalogs.
         """
-        # fetch course catalog for the given catalog id.
         catalog_api = CourseCatalogApiClient(request.user)
         catalog = catalog_api.get_catalog(pk)
-
-        if not catalog:
-            logger.error("Unable to fetch API response for given catalog from endpoint '/catalog/%s/'.", pk)
-            raise NotFound("The resource you are looking for does not exist.")
-
+        self.ensure_data_exists(request, catalog)
```

Comment:
I think the issue here is I am using the same error message to log *and* to throw, but here they use different ones. Which would you recommend to pass in? I don't know if I want to make an extra parameter for differentiating between logging/throwing, to keep things simple & consistent.